lib/sensu/server/process.rb

```diff
@@ -572,6 +575,58 @@ def determine_stale_clients
         end
       end
 
+      # Determine stale checks, those that have not sent an event
+      # in a specified amount of time (thresholds). This method
+      # iterates through the result registry. If the time since
+      # the latest event is equal to or greater than a threshold,
+      # the check `:output` is set to a descriptive message, and
+      # `:status` is set to the appropriate non-zero value, and an
+      # alert is sent. If a check has been sending events, then
+      # no alert is published.
+      def determine_stale_checks
+        @logger.info("determining stale checks")
+        @redis.smembers("clients") do |clients|
+          clients.each do |client_name|
+            @redis.keys("result:" + client_name + ":*") do |results|
+              results.each do |result_name|
+                next if result_name.split(':')[2] == 'keepalive'
+                @redis.get(result_name) do |result_json|
+                  next if result_json.nil?
+                  result = MultiJson.load(result_json)
+                  check_defaults = {
+                    :thresholds => {
```

Comment:
Firstly, checks need to be manually removed through the API if their client config is removed, so in that sense this is no different, and will be handled at the same time as a check is normally removed.

Secondly, I have no strong opinions about the default thresholds being 0 (and therefore disabled) or the same as client keepalives. My argument for the former would be as I said before: it's currently possible for broken checks to be listed as 'OK', which I think is worse than changing the behaviour.

Either way, the functionality is what's important to me :)